tests/fsharp/core/nameof/version46/test.fsx

```diff
@@ -0,0 +1,404 @@
+//<Expects id="FS0039" status="error" span="(100,17)">The value or constructor 'nameof' is not defined.</Expects>
```

Comment:
Should the error be "nameof requires `/langversion:4.7` or greater?"

Reply:
That is the original error message in older versions of the compiler.

Reply:
I don't think we need be this strict, but I'll leave it up to you and @cartermp.  It feels like if we can program a better user experience easily we should.

Reply:
@dsyme, I agree, that we can improve things as we move forward.

Reply:
This should have been covered in the LangVersion RFC, but we need to emit a diagnostic about it not being available in the user's current LangVersion.